### ISSUE TITLE
LDAP Integration Issues

### ISSUE THREAD

**Issue by norwalk350** [Seitanas/kvm-vdi#72]
Hello, again,

I'm having issues getting LDAP integration to work. Sorry if I'm bugging you too much. I'm a sysadmin, not a developer. PHP and other programming languages are not my forte. 

I added the correct information in /var/www/html/kvm-vdi/functions/config.php, but I am getting the following errors when I try to authenticate via LDAP

![image](https://cloud.githubusercontent.com/assets/22620330/20540365/fa8de316-b0c6-11e6-9534-6226a85171fc.png)

The whole error is: PHP Warning:  ldap_bind(): Unable to bind to server: Invalid credentials in /var/www/html/kvm-vdi/functions/functions.php on line 327

I've checked functions.php and everything seemed ok. I checked how the variables correspond to the form values in the client_index.php file. Is there something I'm missing? I've verified my AD server is listening on port 389 and used a test .php file to verify I can successfully bind with AD.

`<?php

$username = 'jtest';
$password = 'jtest';
$server = 'x.x.x.x';
$domain = '@fail.local';
$port = 389;

$connection = ldap_connect($server, $port);
if (!$connection) {
    exit('Connection failed');
}

// Help talking to AD
ldap_set_option($ldap_connection, LDAP_OPT_PROTOCOL_VERSION, 3);
ldap_set_option($ldap_connection, LDAP_OPT_REFERRALS, 0);

$bind = @ldap_bind($connection, $username.$domain, $password);
if (!$bind) {
    exit('Binding failed');
}

// This is where you can do your work
echo 'Hello from LDAP';

ldap_close($ldap_connection);
?>

As always, any help is much appreciated! `

**Comment by Seitanas** [comment]
So are you using AD or LDAP backend for your credential provider?

**Comment by norwalk350** [comment]
I’m using AD as my credential provider.



*From:* Tadas Ustinavičius [mailto:notifications@github.com]
*Sent:* Tuesday, November 22, 2016 4:32 PM
*To:* Seitanas/kvm-vdi
*Cc:* norwalk350; Author
*Subject:* Re: [Seitanas/kvm-vdi] LDAP Integration Issues (#72)



So are you using AD or LDAP backend for your credential provider?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Seitanas/kvm-vdi/issues/72#issuecomment-262372013>, or mute
the thread

**Comment by Seitanas** [comment]
This sounds, that your `$base_dn`  is perhaps misconfigured.
You need to set proper base_dn for your AD connection.
Something like:
`$base_dn = 'OU=People,DC=adserver,DC=domain,DC=com;`

**Comment by norwalk350** [comment]
LDAP seems to be working now. My $base_dn was correct. I verified it with another web application I use. I did some research online. I think adding this line to the functions.php helped:

![image](https://cloud.githubusercontent.com/assets/22620330/20682380/bf25961c-b575-11e6-880a-ae34fa1c8e04.png)

Now, I'm having this issue:

![image](https://cloud.githubusercontent.com/assets/22620330/20682320/8bb6b554-b575-11e6-94ef-f0514d46f81d.png)

When I login as a LDAP user, it creates the account in the mySQL backend, which is good. However, I keep getting the "valuechar" error when I login. I'm also not seeing the clients in the dashboard either.
